### ISSUE TITLE
ExternalPerturbationLangevinIntegrator produces strange energies with `updateParametersInContext`

### ISSUE THREAD

**Issue by gregoryross** [choderalab/openmmtools#173]
I've encountered large discrepancies in calculated energies when using `ExternalPerturbationLangevinIntegrator`. The code below compares protocol work that is accumulated inside `ExternalPerturbationLangevinIntegrator` and outside the integrator using `getState()`. The following example is a model of an NCMC procedure where no perturbation is performed on the system but `updateParametersInContext()` is still called. When `updateParametersInContext()` is called, the different methods of calculating protocol work produce values that differ on the order of 1kJ/mol.

Tagging @jchodera, @bas-rustenburg, @pgrinaway and @maxentile.
``` python
from simtk import openmm, unit
from simtk.openmm import app
import numpy as np

# Using one CPU thread
import os
os.environ['OPENMM_CPU_THREADS'] = '1'

# Long range method
nonbonded_method = 'CutoffPeriodic'

# Creating a waterbox
from openmmtools.testsystems import WaterBox
wbox = WaterBox(box_edge=20.0*unit.angstrom , nonbondedMethod=getattr(app, nonbonded_method))
wbox.system.addForce(openmm.MonteCarloBarostat(1*unit.atmospheres, 300*unit.kelvin))

# Extracting the nonbonded force
non_bonded_force = wbox.system.getForce(2)

# The integrator to perform the equilibrium dynamics
from openmmtools.integrators import ExternalPerturbationLangevinIntegrator
integrator = ExternalPerturbationLangevinIntegrator(temperature=300*unit.kelvin, collision_rate=50.0 / unit.picosecond, timestep=1.0 * unit.femtosecond)

# Creating the context
platform = openmm.Platform.getPlatformByName('CPU')
context = openmm.Context(wbox.system, integrator, platform)
context.setPositions(wbox.positions)

# Running some equilibrium dynamics
integrator.step(100)

# The number of NCMC type iterations and NCMC steps per iteration.
niterations = 20
ncmc_steps = 10

internal_work = np.zeros(niterations)
external_work = np.zeros(niterations)

# Whether to call updateParametersInContext. If True, then assertion below will fail.
update_parameters = True

# A model of NCMC without perturbation but using updateParametersInContext
for i in range(niterations):
    integrator.setGlobalVariableByName('first_step',0)
    integrator.setGlobalVariableByName('protocol_work',0)
    for s in range(ncmc_steps):
        integrator.step(1)
        initial_external_energy = context.getState(getEnergy=True).getPotentialEnergy() / unit.kilojoule_per_mole
        ###---- Not perturbing the system but updating parameters anyway----###
        if update_parameters:
            non_bonded_force.updateParametersInContext(context)
        final_external_energy = context.getState(getEnergy=True).getPotentialEnergy() / unit.kilojoule_per_mole
        integrator.step(1)
    internal_work[i] = integrator.getGlobalVariableByName('protocol_work')
    external_work[i] = final_external_energy - initial_external_energy

print('internal_work: %s' %internal_work)
print('external_work: %s' % external_work)

assert np.all(np.abs(internal_work - external_work) < 1E-1)
```

**Comment by jchodera** [comment]
I've made a few edits to your script above to get it to run.

Note that I believe the number of CPU threads is determined at startup, so I don't think you can control the number of threads via `os.environ` like that. Alternatively, you can set a platform property that controls the number of threads within the Python program.
Currently, the script returns this:
```python
internal_work: [  0.00000000e+00   0.00000000e+00  -1.66125450e+00   0.00000000e+00
   1.94054068e+00   0.00000000e+00   0.00000000e+00   8.68733668e+00
   0.00000000e+00   1.43813417e+00   0.00000000e+00   0.00000000e+00
  -3.41660889e+00   0.00000000e+00  -5.75446671e+00   0.00000000e+00
   0.00000000e+00   1.85193903e-04   0.00000000e+00  -1.35276045e+00]
external_work: [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.]
```
However, removing the barostat gives perfect agreement:
```
mski1776:tests choderaj$ python test_external.py 
internal_work: [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.]
external_work: [ 0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.
  0.  0.]
```

**Comment by jchodera** [comment]
Fixed by #175
